apiserver/facades/client/client/filtering.go

```diff
@@ -101,15 +101,15 @@ func BuildPredicateFor(patterns []string) Predicate {
 		default:
 			panic(errors.Errorf("expected a machine or an applications or a unit, got %T", i))
 		case *state.Machine:
-			shims, err := buildMachineMatcherShims(i.(*state.Machine), patterns)
+			shims, err := buildMachineMatcherShims(i.(*state.Machine), leaders, patterns)
 			if err != nil {
 				return false, err
 			}
 			return or(shims...)
 		case *state.Unit:
-			return or(buildUnitMatcherShims(i.(*state.Unit), patterns)...)
+			return or(buildUnitMatcherShims(i.(*state.Unit), leaders, patterns)...)
 		case *state.Application:
-			shims, err := buildApplicationMatcherShims(i.(*state.Application), patterns...)
+			shims, err := buildApplicationMatcherShims(i.(*state.Application), leaders, patterns...)
```

Comment:
This method takes a variadic `patterns`, which is always splatted from a slice.

Change that and you might as well be receiving `matcherParams` higher up the stack instead of passing both patterns and leaders and only building it `buildUnitMatcherShims`.